plugin/gcp/cloudFunction.go

```diff
@@ -0,0 +1,55 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/puppetlabs/wash/plugin"
+	"google.golang.org/api/cloudfunctions/v1"
+)
+
+type cloudFunction struct {
+	plugin.EntryBase
+	service cloudFunctionsProjectService
+}
+
+func newCloudFunction(function *cloudfunctions.CloudFunction, service cloudFunctionsProjectService) *cloudFunction {
+	cf := &cloudFunction{
+		EntryBase: plugin.NewEntry(function.Name),
```

Comment:
Looks like cloud function names are of the form `project/<project_id>/location/<region>/functions/<function_name>`. Given this, how should we organize them? I'm thinking `<region>::<function_name>`.

Reply:
VM instances are also in a specific region/zone. As long as name uniqueness is enforced across zones I didn't put it in.